stdlib/public/core/ArrayBufferProtocol.swift

```diff
@@ -89,6 +89,7 @@ public protocol _ArrayBufferProtocol : MutableCollection {
   /// over the underlying contiguous storage.
   ///
   /// - Precondition: Such contiguous storage exists or the buffer is empty.
+  @discardableResult
   mutating func withUnsafeMutableBufferPointer<R>(
```

Comment:
These two cases don't seem right to me: if you're passing a closure that returns something, withUnsafeMutableBufferPointer shouldn't allow you to ignore it accidentally.  I'd suggest removing the attribute from these two.